test/web3/schema.json

```diff
@@ -0,0 +1,163 @@
+      "gasPrice": "Q",
+      "gas": "Q",
+      "input": "D"
+    },
+    "Receipt": {
+      "__required": [],
+      "transactionHash": "D32",
+      "transactionIndex": "Q",
+      "blockHash": "D32",
+      "blockNumber": "Q",
+      "cumulativeGasUsed": "Q",
+      "gasUsed": "Q",
+      "contractAddress": "D20",
+      "logs": ["FilterChange"]
+    },
+  
+    "FilterChange": {
+      "__required": [],
+      "removed": "B",
+      "logIndex": "Q",
+      "transactionIndex": "Q",
+      "transactionHash": "D32",
+      "blockHash": "D32",
+      "blockNumber": "Q",
+      "address": "D20",
+      "data": "Array|DATA",
+      "topics": ["D"]
+    },
+    "methods1": {
+      "eth_blockNumber": ["params1", "Q"],
+      "eth_gasPrice": ["params1", "Q"],
+      "eth_newBlockFilter": ["params1", "Q"],
+      "eth_hashrate": ["params1", "Q"],
+      "eth_newPendingTransactionFilter": ["params1", "Q"],
+      "eth_getUncleCountByBlockHash": ["params6", "Q", 1],
+      "eth_getBlockTransactionCountByHash": ["params6", "Q", 1],
+      "eth_getTransactionCount": [["params9", "params7"], "Q", 1, 2],
+      "eth_getBalance": [["params9", "latest"], "Q", 1, 2],
+      "eth_estimateGas": [["EstimateTransaction"], "Q", 1],
+      "eth_getUncleCountByBlockNumber": [["params7"], "Q", 1],
+      "eth_newFilter": [["FilterNewFilter"], "Q", 1],
+      "eth_getBlockTransactionCountByNumber": [["latest"], "Q", 1]
+    },
+    "methods2": {
+      "eth_uninstallFilter": ["params2", "B", 1],
+      "eth_mining": ["params1", "B"],
+      "eth_submitWork": ["submitworkparams", "B", 3],
+      "eth_syncing": ["params1", "B|EthSyncing"]
+    },
+    "methods3": {
+      "eth_getTransactionByHash": ["params3", "Transaction", 1],
+      "eth_getTransactionByBlockHashAndIndex": [
+        ["params4", "Q"],
+        "Transaction",
+        2
+      ],
+      "eth_getTransactionByBlockNumberAndIndex": [
+        ["params7", "0x0"],
+        "Transaction",
+        2
+      ]
+    },
+    "methods4": {
+      "eth_getUncleByBlockHashAndIndex": ["params4", "Block", 1],
+      "eth_getUncleByBlockNumberAndIndex": ["params5", "Block", 2],
+      "eth_getBlockByHash": [["params", false], "Block", 2],
+      "eth_getBlockByNumber": [["params7", false], "Block", 2]
+    },
+  
+    "methods5": {
+      "eth_getWork": ["params1", ["D"]],
+      "eth_call": [["EstimateTransaction", "latest"], "D", 1, 2],
+      "eth_getStorageAt": ["getstorageatparams", "D", 2, 2],
+      "eth_sendTransaction": [["SendTransaction"], "D", 1],
+      "eth_getCode": ["getcodeparams", "D", 1, 2]
+    },
+    "methods6": {
+      "eth_getFilterChanges": ["params2", ["FilterChange"], 1],
+      "eth_getLogs": [["FilterGetLogs"], ["FilterChange"], 1]
+    },
+    "methods": {
+      "eth_accounts": ["params1", ["D20"]],
+      "eth_protocolVersion": ["params1", "S"],
+      "eth_sendRawTransaction": [["params8"], "D32", 1],
+      "eth_getTransactionReceipt": ["params3", "Receipt", 1]
+    }
+  }
+  
```

Comment:
Also, it might be good to group the methods and params objects in some other object, for example:
```
"methods": {
    "blockMethods": {
        "getBlockByHash":
...
```